sqrl-openai/src/main/java/com/datasqrl/openai/FunctionExecutor.java

```diff
@@ -4,32 +4,44 @@
 import org.apache.flink.table.functions.FunctionContext;
 
 import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 import static com.datasqrl.openai.RetryUtil.executeWithRetry;
 
 public class FunctionExecutor {
 
     private final FunctionMetricTracker metricTracker;
+    private final ExecutorService executorService;
 
     public FunctionExecutor(FunctionContext context, String functionName) {
         this.metricTracker = new FunctionMetricTracker(context, functionName);
+        this.executorService = Executors.newFixedThreadPool(10);
```

Comment:
Unfortunately, I couldn’t find a way to share the executor. The most optimal approach would be to borrow one from the context, but as far as I know, such a feature doesn’t exist. Would it make sense to make the pool size configurable via an environment variable, specific to the function? For instance, something like FUNCTION_NAME.toUpperCase() + "_POOL_SIZE".

Reply:
That's fine, we can have one threadpool per function and use a single environment variable ASYNC_FUNCTION_THREAD_POOL_SIZE to control it across all functions. We can later decide if we need to make it function specific.